src/main/java/co/rsk/bitcoinj/script/RedeemScriptValidator.java

```diff
@@ -219,8 +229,35 @@ protected static List<ScriptChunk> removeOpCheckMultisig(Script redeemScript) {
         return redeemScript.getChunks().subList(0, redeemScript.getChunks().size() - 1);
     }
 
-    protected static boolean isOpN(ScriptChunk chunk) {
-        return chunk.isOpCode() &&
-            chunk.opcode >= ScriptOpCodes.OP_1 && chunk.opcode <= ScriptOpCodes.OP_16;
+    public static int decodeN(ScriptChunk chunk) {
+        if (chunk.isOpCode()) {
+            return Script.decodeFromOpN(chunk.opcode);
+        }
+
+        if (chunk.isPushData() && !isNull(chunk.data) && chunk.data[0] >= 1) {
```

Comment:
Isn't `isPushDataNumber` already doing this validation?

Reply:
Updated